contracts/Removal.sol

```diff
@@ -23,22 +23,29 @@ contract Removal is
 {
   using SafeMathUpgradeable for uint256;
 
-  struct Vintage {
-    address supplier;
-    uint16 vintage;
-    // todo: location
-    // todo: methodology
-    // todo: supplier name
-  }
-
   struct BatchMintRemovalsData {
     address marketAddress;
     bool list;
   }
 
-  mapping(uint256 => Vintage) private _vintages;
+  uint256 constant _BITS_PER_BYTE = 8;
+  uint256 constant _ADDRESS_FIELD_LENGTH = 20;
+  uint256 constant _ADDRESS_OFFSET = 12;
+  uint256 constant _PARCEL_ID_FIELD_LENGTH = 5;
+  uint256 constant _PARCEL_ID_OFFSET = 7;
+  uint256 constant _VINTAGE_FIELD_LENGTH = 2;
+  uint256 constant _VINTAGE_OFFSET = 5;
+  uint256 constant _COUNTRY_CODE_FIELD_LENGTH = 2;
+  uint256 constant _COUNTRY_CODE_OFFSET = 3;
+  uint256 constant _STATE_CODE_FIELD_LENGTH = 2;
+  uint256 constant _STATE_CODE_OFFSET = 1;
+  uint256 constant _METHODOLOGY_DATA_FIELD_LENGTH = 1;
+  uint256 constant _METHODOLOGY_DATA_OFFSET = 0;
+
   uint256 private _latestTokenId;
   string public name; // todo why did I add this
+  mapping(uint256 => uint256) public orderedTokenIds;
+  mapping(uint256 => bool) private _tokenIds;
```

Comment:
these maps are for what we talked about regarding future-proofing our ability to iterate sequentially over token ids that are now of arbitrary ordering. the second map keeps track of token ids for existence so we can prevent duplications. this seemed like the better approach than trying to call `balanceOf` for the tokenId because that function requires an `address` argument, and we don't care about who may or may not have a balance of that token, we just need to know if its id exists at all.  And we don't have access to the underlying balances mapping from this contract so we keep track ourselves. maybe there's a better way

Reply:
I think this is fine.  Some take it or leave it naming ideas:

`indexToTokenId` (easier to intuit which the ids are on)
`_tokenIdExists` (more explicit about the bool value means)

Reply:
love it